corehq/motech/models.py

```diff
@@ -1,20 +1,56 @@
 import re
 
+from django.contrib.postgres.fields import JSONField
 from django.core.serializers.json import DjangoJSONEncoder
 from django.db import models
 
 import jsonfield
 
 from corehq.motech.const import (
     ALGO_AES,
-    BASIC_AUTH,
-    DIGEST_AUTH,
+    AUTH_TYPES,
     OAUTH1,
+    OAUTH2_BEARER,
     PASSWORD_PLACEHOLDER,
 )
 from corehq.motech.utils import b64_aes_decrypt, b64_aes_encrypt
 
 
+class ApiAuthSettings(models.Model):
```

Comment:
do you think this really needs to be dynamic (as opposed to being specified in code)?

Reply:
You mean hardcoding for DHIS2, ipswitch / MOVEit Automation, and any other APIs we encounter? I think if the number of different APIs we deal with is small, that would be fine. ... Perhaps this is a YAGNI instance, where we can hardcode until we need to move to a model.